src/wp-includes/functions.php

```diff
@@ -4266,6 +4266,40 @@ function wp_check_jsonp_callback( $callback ) {
 	return 0 === $illegal_char_count;
 }
 
+/**
+ * Reads and decodes a JSON file.
+ *
+ * @since 5.9.0
+ *
+ * @param string $filename Path to the JSON file.
+ * @param array  $options  {
+ *     Optional. Options to be used with `json_decode()`.
+ *
+ *     @type bool associative Optional. When `true`, JSON objects will be returned as associative arrays.
+ *                            When `false`, JSON objects will be returned as objects.
+ * }
+ *
+ * @return mixed Returns the value encoded in JSON in appropriate PHP type.
+ *               `null` is returned if the file is not found, or its content can't be decoded.
+ */
+function wp_json_file_decode( $filename, $options = array() ) {
```

Comment:
Are there other places in core where such a function would be useful?

Otherwise, not sure such a thin wrapper around `json_decode( file_get_contents( $filename ), true )` is useful enough.

Reply:
I don't feel strongly about adding `wp_json_file_decode`. It's the same logic that could be unified between `block.json`, `theme.json`, and the related i18n schema handling. Error handling would be useful for `block.json` since it's provided by plugins, but I could inline the logic.

Reply:
Just saying that now I added translations support and path normalization to `wp_json_file_decode`, it makes it more appealing to keep it as a shared utility. 😄